src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -7,22 +7,36 @@
 using System.Reflection;
 using System.Runtime.Loader;
 
-
-//using Roslyn.Utilities;
-
-//using static Microsoft.CodeAnalysis.AnalyzerAssemblyLoadUtils;
-
 namespace Microsoft.Build.Shared
 {
-    /// Core CLR compatible wrapper for loading analyzers.
+    /// <summary>
+    /// CoreCLR-compatible wrapper for loading task assemblies.
+    /// </summary>
+    /// <remarks>
+    /// This type defines a new <see cref="AssemblyLoadContext"/>. This isn't really what we want to
+    /// do, as we aren't trying to isolate our assembly loads from the default context. All we want
+    /// to do is add support for loading task assemblies and their dependencies, which we assume are
+    /// located immediately next to them in the file system. Ideally we wouldn't derive from
+    /// <see cref="AssemblyLoadContext"/> at all, but simply hook the <see cref="AssemblyLoadContext.Resolving"/>
+    /// event to handle finding the dependencies. For the moment, however, that approach is blocked
+    /// by https://github.com/dotnet/coreclr/issues/5837.
+    /// </remarks>
     internal sealed class CoreClrAssemblyLoader : AssemblyLoadContext
     {
         private readonly Dictionary<string, Assembly> _pathsToAssemblies = new Dictionary<string, Assembly>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<string, Assembly> _namesToAssemblies = new Dictionary<string, Assembly>();
         private readonly List<string> _dependencyPaths = new List<string>();
         private readonly object _guard = new object();
+
+        private readonly AssemblyLoadContext _defaultContext;
```

Comment:
I think that was a holdover from when I was trying to hook the `Resolving` event.

It may also be possible that the default `AssemblyLoadContext` could change, and you could argue that we're better off using the same one every time rather than (possibly) a different one, so we should capture it up front. I'm not sure how likely that scenario is.